src/utils/isCustomPropertySet.js

```diff
@@ -0,0 +1,9 @@
+/**
+ * Check whether a property is a custom properties
+ *
+ * @param {string} customPropertySet
+ * @return {boolean} If `true`, property is a custom properties
+ */
+export default function (customPropertySet) {
+  return (customPropertySet.slice(0, 2) === "--" && customPropertySet.slice(-1) === ":")
+}
```

Comment:
@jeddy3 first custom property set not have `;` in `before`. Ref: https://github.com/stylelint/stylelint/issues/1830#issuecomment-246736506, we can always look to the next node, and for the last node look at the parent `after`. If this logic nobody believes hard i am rewrite this.